### PR TITLE
Measure and dimension visibility selection in dashboard

### LINKED ISSUE #308: Dashboard UX: implement measures & dimension visibility menus
The metrics explorer will need to implement menus that enable the user to change the visibility. A fast-path here would be to use the existing menu components in`src/lib/components/menu` and look at examples of the menus in action.

These menus can be implement in a dev route, or done directly in the existing dev route for the metrics explorer (shoudl be called `dev/leaderboard` or `dev/explore` as of this issue filing).

Some features:
- it should be a button that creates a floating menu component
- it should create a list of menu items that either have a checkbox or don't have a checkbox. If there is no checkbox icon, work with marissa to get one in figma and add it to the repository.
- optionally, it would be nice if we had quick fuzzy text searching of the menu items. If this is infeasible, it's better to land the code without it, but please file a follow-on issue for that.

### PR DESCRIPTION
**Pull request by bcolloran**
will fix #308 
see https://www.notion.so/rilldata/Ability-to-hide-measures-and-dimensions-from-the-dashboard-f416bfaddda74e5285ec8bc33161a2c0

Note from @hamilton: this PR will only implement the components for showing & hiding the measure dimensions. As of merging, we'll still need to do the following:
- ensure a unique `name` field exists for all measures & dimensions. This is backend work on the runtime.
- capture state for the shown / hidden measures & dims in the dashboard state (and then the actual stateful URL).
- auto-generate dashboards should template in a `name` field for each dimension (once we have `name` in for dimensions).

cc @begelundmuller for backend parts

### CODE REVIEW

**Comment by bcolloran** [comment]
@hamilton this is ready for an experience pass

@AdityaHegde, can you do code review and also wire up whatever is needed to capture the visibility state of the measures and dimensions in the URL? Hamilton said that he thought that would be pretty quick and easy for you, lmk if I can help

File: web-common/src/features/dashboards/leaderboard/LeaderboardMeasureSelector.svelte
Comment (hamilton) [inline web-common/src/features/dashboards/leaderboard/LeaderboardMeasureSelector.svelte:104]: I found this text a bit confusing.

`All dimensions sorted by total sales` reads like I sort the leaderboard cards themselves, not the values within them. Not sure I have a solution for this.

File: web-common/src/features/dashboards/leaderboard/LeaderboardMeasureSelector.svelte
Comment (hamilton) [inline web-common/src/features/dashboards/leaderboard/LeaderboardMeasureSelector.svelte:94]: Could we possibly line up these selectors with the text in the first leaderboard in the x direction?

Screen Shot 2023-04-11 at 6 00 37 AM

File: web-common/src/features/dashboards/leaderboard/LeaderboardMeasureSelector.svelte
Comment (hamilton) [reply web-common/src/features/dashboards/leaderboard/LeaderboardMeasureSelector.svelte:104]: We could change this to "showing top values by [selection]"

**Review by hamilton — CHANGES_REQUESTED** [review]
Thanks for the PR! The functionality is great and the code is very clean. 

I left a few comments on both design and implementation below. The most obvious one is moving away from the bitmap. I think this may cause some issues down the line as noted in the comment.

If you can fix that first, then Aditya will jump in and add state to this when he begins his workday tomorrow.

File: web-common/src/components/searchable-filter-menu/SearchableFilterDropdown.svelte
Comment (hamilton) [inline web-common/src/components/searchable-filter-menu/SearchableFilterDropdown.svelte:50]: I think we're not 100% consistent here, but I tend to prefer `select-all` kebab casing for component events, and this is what's most dominant in the code I think.

File: web-common/src/components/data-graphic/marks/MultiMetricMouseoverLabel.svelte
Comment (bcolloran) [inline web-common/src/components/data-graphic/marks/MultiMetricMouseoverLabel.svelte:137]: added this condition to fix a bug I was hitting where `container` was not present at the time the setTimeout was called

**Comment by bcolloran** [comment]
@hamilton all comments addressed, though I left in FIXMEs for the entity keys. If that is expected to land next week, recommend we hold off on final review on this branch until that PR lands, at which point I can  merge in those changes and clear the FIXMEs.

**Comment by bcolloran** [comment]
@hamilton -- updating my previous recommendation, I think we should merge this PR as-is (pending any changes needed from review) rather than waiting for an entity key solution to fully land. This PR does add nice incremental functionality in it's current state, and even if the serialization to stateful URL lands in a subsequent PR, I think this will still an improvement.

Once @AdityaHegde and/or @begelundmuller have finalized the entity key it should be a simple matter to swap out the items used as keys in `visibleMeasureKeys` and `visibleDimensionKeys` in `dashboard-stores.ts`.

@hamilton  what do you think? Can you do another review pass and merge?

**Comment by bcolloran** [comment]
@AdityaHegde  can you review this PR?

We should land this functionality, then you can follow up with a second PR to make sure the visibility state is synced in the URL -- this will be valuable even without that final bit of functionality, and that should be a fast and easy follow up hopefully.

**Review by hamilton — APPROVED** [review]

**Pull request merged**
